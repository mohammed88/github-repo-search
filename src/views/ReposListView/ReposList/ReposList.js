import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Spacing from '../../../ui/styles/spacing';
import Item from '../../../components/Item/';
import ItemLoading from '../../../components/ItemLoading/';

const RepoItem = styled(Item)`
	margin-bottom: ${Spacing.medium};
`;
export const RepoItemLoading = styled(ItemLoading)`
	margin-bottom: ${Spacing.medium};
`;

function ReposList({repos, isLoading, owner, repo}) {
	return (
		<React.Fragment>
			{repos.map(repo =>
				<RepoItem
					key={repo.id}
					repo={repo}
					detailsUrl={`/repo/${repo.full_name}`}
				/>
			)}
			{isLoading &&
				Array(10).fill(null).map((v, i) => <RepoItemLoading key={i} />)
			}
			{repos.length === 0 && !isLoading &&
				<div>No results found.</div>
			}
		</React.Fragment>
	);
}

ReposList.propTypes = {
	repos: PropTypes.array.isRequired, //TODO: type check each item
	isLoading: PropTypes.bool,
};

ReposList.defaultProps = {
	isLoading: false,
}

export default ReposList;