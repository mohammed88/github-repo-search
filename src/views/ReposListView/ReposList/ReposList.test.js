import React from 'react';
import {shallow} from 'enzyme';

import ReposList, {RepoItemLoading} from './ReposList';

describe('ReposList', () => {
	it('should render', () => {
		const wrapper = shallow(<ReposList repos={[]} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should render repos placeholder when loading', () => {
		const wrapper = shallow(<ReposList repos={[]} isLoading={true} />);

		expect(wrapper.find(RepoItemLoading).length).toBe(10);
	});
});