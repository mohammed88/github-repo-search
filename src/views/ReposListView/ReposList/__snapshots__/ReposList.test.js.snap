// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposList should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReposList
    isLoading={false}
    repos={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        Array [],
        false,
        <div>
          No results found.
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No results found.",
        },
        "ref": null,
        "rendered": "No results found.",
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          Array [],
          false,
          <div>
            No results found.
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No results found.",
          },
          "ref": null,
          "rendered": "No results found.",
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
